src/viam/sdk/common/proto_utils.hpp

```diff
@@ -0,0 +1,61 @@
+/// @file common/proto_utils.hpp
+///
+/// @brief Utils that require generated proto includes. These should be #included
+/// in cpp implementation files, but not in wrapper headers consumed by third party code.
+#pragma once
+
+#include <viam/api/common/v1/common.pb.h>
+
+namespace viam {
+namespace sdk {
+
+/// @brief Copies elements from a protobuf repeated pointer array into a std::vector. Src type
+/// must be implicitly convertible to Dst (probably via operator on Src).
+template <typename Src, typename Dst>
+void vecToRepeatedPtr(const std::vector<Src>& vec, google::protobuf::RepeatedPtrField<Dst>& dest) {
+    dest.Clear();
+    dest.Reserve(vec.size());
+    for (auto& x : vec) {
+        *dest.Add() = x.to_proto();
+    }
+}
+
+/// @brief Non-member to_proto() version. (necessary for moving generated types out of wrapper
+/// headers). Template param F is a function that converts from Src to Dst.
+template <typename Src, typename Dst>
```

Comment:
explain this one -- this is better because it would select this specialization and return a static assert error, rather than giving a long 'no matching template' error?

Reply:
I think, as written, yours will only accept function pointers. I could be wrong, I'd need to test it out on godbolt.